march_hardware/src/Encoder.cpp

```diff
@@ -1,122 +1,138 @@
+  const double difference = std::abs(encoder_range_of_motion - range_of_motion) / encoder_range_of_motion;
+  if (difference > Encoder::MAX_RANGE_DIFFERENCE)
+  {
+    ROS_WARN("Difference in range of motion of %.2f%% exceeds %.2f%%\n"
+             "Encoder range of motion = %f rad\n"
+             "Limits range of motion = %f rad",
+             difference * 100, Encoder::MAX_RANGE_DIFFERENCE * 100, encoder_range_of_motion, range_of_motion);
+  }
 }
 
-double Encoder::getAngleRad(uint8_t ActualPositionByteOffset)
+double Encoder::getAngleRad(uint8_t actual_position_byte_offset) const
 {
-  return IUtoRad(getAngleIU(ActualPositionByteOffset));
+  return this->toRad(this->getAngleIU(actual_position_byte_offset));
 }
 
-int32_t Encoder::getAngleIU(uint8_t ActualPositionByteOffset)
+int32_t Encoder::getAngleIU(uint8_t actual_position_byte_offset) const
 {
-  if (this->slaveIndex == -1)
+  if (this->slave_index_ == -1)
   {
     ROS_FATAL("Encoder has slaveIndex of -1");
   }
-  union bit32 return_byte = get_input_bit32(this->slaveIndex, ActualPositionByteOffset);
+  union bit32 return_byte = get_input_bit32(this->slave_index_, actual_position_byte_offset);
   ROS_DEBUG("Encoder read (IU): %d", return_byte.i);
   return return_byte.i;
 }
 
-int32_t Encoder::RadtoIU(double rad)
-{
-  return static_cast<int32_t>(rad * totalPositions / (2 * M_PI) + zeroPositionIU);
-}
-
-double Encoder::IUtoRad(int32_t iu)
+int32_t Encoder::fromRad(double rad) const
 {
-  return (iu - zeroPositionIU) * 2 * M_PI / totalPositions;
+  return (rad * this->total_position_ / PI_2) + this->zero_position_iu_;
 }
 
-void Encoder::setSlaveIndex(int slaveIndex)
+double Encoder::toRad(int32_t iu) const
 {
-  this->slaveIndex = slaveIndex;
+  return (iu - this->zero_position_iu_) * PI_2 / this->total_position_;
 }
 
-int Encoder::getSlaveIndex() const
+bool Encoder::isWithinHardLimitsIU(int32_t iu) const
 {
-  return this->slaveIndex;
+  return (iu > this->lower_limit_iu_ && iu < this->upper_limit_iu_);
 }
 
-bool Encoder::isWithinHardLimitsIU(int32_t positionIU)
+bool Encoder::isWithinSoftLimitsIU(int32_t iu) const
 {
-  return (positionIU > this->lowerHardLimitIU && positionIU < this->upperHardLimitIU);
+  return (iu > this->lower_soft_limit_iu_ && iu < this->upper_soft_limit_iu_);
 }
 
-bool Encoder::isWithinSoftLimitsIU(int32_t positionIU)
+bool Encoder::isValidTargetIU(int32_t current_iu, int32_t target_iu) const
 {
-  return (positionIU > this->lowerSoftLimitIU && positionIU < this->upperSoftLimitIU);
-}
-
-bool Encoder::isValidTargetIU(int32_t currentIU, int32_t targetIU)
-{
-  if (this->isWithinSoftLimitsIU(targetIU))
+  if (this->isWithinSoftLimitsIU(target_iu))
   {
     return true;
   }
 
-  if (currentIU >= this->getUpperSoftLimitIU())
+  if (current_iu >= this->upper_soft_limit_iu_)
   {
-    return (targetIU <= currentIU) && (targetIU > this->getLowerSoftLimitIU());
+    return (target_iu <= current_iu) && (target_iu > this->lower_soft_limit_iu_);
```

Comment:
This can be done more efficiently with the two (instead of three) conditions
`if (target_iu >= this->upper_soft_limit_iu_)
{
  return (target_iu <= current_iu)
}`

same for the statement below. You could then even remove the above if (this->isWithinSoftLimitsIU(target_iu))
  {
    return true;
  }
and simply return true if both of the if-statements suggested here return true.